src/components/Select/index.tsx

```diff
@@ -28,11 +39,13 @@ const StyledGrommetSelect = styled(
 `;
 
 // Set button to occupy parent's width, and a default width.
-const Container = styled.div`
+const Container = styled.div(
+	(props: { theme: any; invalid?: boolean }) => `
```

Comment:
You don't need to do this, keep is as `styled.div''` instead. Then, it will be `border-color: ${getBorderColor}` or `border-color: ${props => props.invalid ? theme.colors.danger.main : 'initial'}`

Reply:
Yep sorry @sradevski, it makes perfect sense, it was to avoid the `Property 'invalid' does not exist on type 'IntrinsicAttributes & Pick<Pick<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>` error (related to this: https://github.com/styled-components/styled-components/issues/630), but can be solved in a better way with an interface.